src/text-editor-presenter.coffee

```diff
@@ -994,6 +1012,20 @@ class TextEditorPresenter
 
       regions
 
+  setOverlayDimensions: (decorationId, itemWidth, itemHeight, contentMargin) ->
```

Comment:
I'm caching the overlay dimensions in the state. Abuse? I feel it is cleaner than keeping track of two structures for the overlays.